### ISSUE TITLE
doesn't work on windows nt 4.0

### ISSUE THREAD

**Issue by MakaronBalalaikaFAN** [JHRobotics/mesa9x#2]
The procedure entry point CreateTool32Snapshot could not be located in the dynamic link libary KERNEL32.dll

**Comment by lorn10** [comment]
Yes, that is to be absolutely expected. This stuff here is made for **DOS based** Windows versions like 95, 98 and Me. :wink:

**Comment by GenSayer** [comment]
This is a hack, but you can make it work on Windows NT 4.0 and 3.51 by removing lines 81-108 from stw_tls.c

![Capture](https://github.com/JHRobotics/mesa9x/assets/64618080/abf91680-b85e-4336-90f5-92fd274df6c1)

Windows NT 4.0 can use both 95 and 98/Me versions, while Windows NT 3.51 can only use the 95 version. While it compiles and works fine and stable for me, I cannot guarantee that you won't run into issues with prolonged use. If anyone is interested, I can start making NT4/3.51 builds.

**Comment by JHRobotics** [comment]
Thanks @GenSayer!

This code seems to be important only if multiple threads are using for rendering and if opengl32.dll is loaded late. I tried to solve this by dynamic linking, so `CreateToolhelp32Snapshot` is loaded on runtime and if only if exists it is used.

I currently haven't any NT 3.51 or 4.0 instance. As young, I have heard horror stories about these systems, so I scare of them :-)

I also created binaries from last commit (`1b5392f2f0f8c200a53125344c47fb25359e8610`), I'll be glad if someone tries them and report if they're usable.

Download here:
https://github.com/JHRobotics/mesa9x/releases/tag/v23.1.6.41

**Comment by LeoKids** [comment]
I've tested and works fine...

**Comment by ruthan** [comment]
Could you write some tutorial how to do it? 
Works fine, more details about tested games etc would help.

**Comment by LeoKids** [comment]
> Could you write some tutorial how to do it? Works fine, more details about tested games etc would help.

I've tested Classicube and welp it works fine when i ran it on NT 4.0,using  VirtualBox.

**Comment by ruthan** [comment]
Well test more older games if you can..

A few years back i made big pack 320 of Windows 98 game demos, at least some of them would work on Windows NT4 and Windows 95 and they are completely free and legal and except very need ones they take not much disk space.

**Comment by LeoKids** [comment]
Is there even a list of OpenGL games that work on NT 4.0?

**Comment by LeoKids** [comment]
> Well test more older games if you can..
> 
> A few years back i made big pack 320 of Windows 98 game demos, at least some of them would work on Windows NT4 and Windows 95 and they are completely free and legal and except very need ones they take not much disk space.

Can you show me the big pack  320?

**Comment by ruthan** [comment]
I through that inserted links to archive.org above so once again.. I hope that not someone removed it, its completely legal.. These were distributed for free as trial before buying, it was nice thing in 1990 and in early 2000.

https://archive.org/details/win-98-games-a-d.-7z
https://archive.org/details/win-98-games-e-j.-7z
https://archive.org/details/win-98-games-k-o.-7z
https://archive.org/details/win-98-games-p-z.-7z

List well there is something, but we need more games to test, there also new Direct3D for NT4 project and DXGL is now ported to NT4 too, but it needs OpenGL 2.0 card - Geforce 6xxx and maybe, maybe some professional ATI cards.
